Apps/phfireeyeetp/fireeyeetp_connector.py

```diff
@@ -0,0 +1,1500 @@
+        ret_val, response = self._paginator(endpoint, action_result, data, method="post")
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        # Add the response into the data section
+        action_result.add_data(response)
+
+        # Return success, no need to set the message, only the status
+        # BaseConnector will create a textual message based off of the summary dictionary
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_on_poll(self, param):
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # pudb.set_trace()
+
+        # Parameters for the API
+        data = {}
+
+        # This is the maximum results per request we can get
+        data['size'] = 200
+
+        # Get config
+        config = self.get_config()
+
+        try:
+            # Get the endtime from Phantom which is when the action was ran
+            timestamp = datetime.utcfromtimestamp(param.get(phantom.APP_JSON_END_TIME) / 1000.0)
+        except:
+            self.debug_print("'end_time' in Phantom could not be converted correctly. Use alternative time equal to datetime.utcnow()")
+        else:
+            timestamp = datetime.utcnow()
+
+        # If it is a manual poll or first run
+        if self.is_poll_now() or self._state.get('first_run', True):
+
+            try:
+                # Check the 'container_count' parameter
+                # If container count is not present just get 1
+                ret_val, limit = self._validate_integer(action_result, param.get(phantom.APP_JSON_CONTAINER_COUNT, 1), CONTAINER_COUNT_KEY)
+                if phantom.is_fail(ret_val):
+                    return action_result.get_status()
+            except:
+                return action_result.set_status(phantom.APP_ERROR, "The maximum containers is invalid")
+
+            date = (timestamp - timedelta(minutes=15)).strftime("%Y-%m-%dT%H:%M:%S.000")
+            data['fromLastModifiedOn'] = date
+
+        # If it is a scheduled poll, ingest from last_ingestion_time
+        else:
+            try:
+                # Get the ingestion interval
+                # If interval is not present just get the last 15 minutes
+                interval_mins = int(config.get('ingest', {}).get('interval_mins', 15))
+            except:
+                return action_result.set_status(phantom.APP_ERROR, "Ingestion interval is invalid")
+
+            # Try to get the last_ingestion_time from the state file
+            # If not get the last x minutes which is determined by the interval
+            date = self._state.get('last_ingestion_time', datetime.utcnow() - timedelta(minutes=interval_mins))
+            data['fromLastModifiedOn'] = datetime.strptime(date, "%Y-%m-%dT%H:%M:%S.000")
+
+        endpoint = FIREETEETP_LIST_ALERTS_ENDPOINT
+
+        # make rest call
+        ret_val, response = self._paginator2(endpoint, action_result, data, limit=limit, method="post")
```

Comment:
I think I had a fix for this and I may have forgot to push it to my repo. I no longer have access to that code to be able to push it. Since the maximum number of events we can get back per request is 200 I would say set limit = data['size'] which will enforce that limit.

Reply:
I think I had a fix for this and I may have forgot to push it to my repo. I no longer have access to that code to be able to push it. Since the maximum number of events we can get back per request is 200 I would say set limit = data['size'] which will enforce that limit.

Reply:
Got it. We can add that in the else block.

Reply:
Fixed